news/2020-02-03-Micro_Profile_OpenTracing_Comes_To_WildFly.adoc

```diff
@@ -0,0 +1,123 @@
+= Micro Profile OpenTracing comes to WildFly =
+ehsavoie
+2020-02-17
+:revdate: 2020-02-03 08:00:00 -0600
+:awestruct-tags: [wildfly, microprofile, configuration, opentracing]
+:awestruct-layout: blog
+:source-highlighter: coderay
+:encoding: utf-8
+:lang: en
+:awestruct-description: How to configure and use Micro Profile OpenTracing within WildFly
+
+Until WildFly 19 you could use MPOT (Micro Profile OpenTracing) to trace your application using environment variables. With wildFly 19 you can now configure several Jaeger Tracers to be used in your applications.
+
+== Installing Jaeger
+
+Let's start a jaeger instance using docker :
+[source,bash]
+--
+docker run -d --name jaeger \
+  -p 16831:6831/udp \
+  -p 5778:5778 \
+  -p 14268:14268 \
+  -p 16686:16686 \
+  jaegertracing/all-in-one:1.16
+--
+
+Now, you can navigate to link:http://localhost:16686[http://localhost:16686] to access the Jaeger UI.
+
+[NOTE]
+====
+For demonstration purpose we have mapped the port *6831* to *16831*.
+====
+
+== Configuring the OpenTracing subsystem
+
+You can use either UDP or TCP to send your log spans to Jaeger.  Note that those configurations are *exclusive*, so if yo configure an TCP endpoint then Jaeger won't use the UDP configuration.
+
+=== Using UDP
+
+First we define an outbound socket binding towards the Jaeger tracer.
+[source,ruby]
+--
+[standalone@localhost:9990 /] /socket-binding-group=standard-sockets/remote-destination-outbound-socket-binding=jaeger:add(host=localhost, port=16831)
+{"outcome" => "success"}
+--
+
+Now we can define our MPOT tracer configuration:
+[source,ruby]
+--
+[standalone@localhost:9990 /] /subsystem=microprofile-opentracing-smallrye/jaeger-tracer=jaeger-demo:add(sampler-type=const, sampler-param=1, reporter-log-spans=true, sender-binding=jaeger)
+{"outcome" => "success"}
+--
+
+=== Using TCP
+
+If you want to use TCP instead of UDP you need to configure the *sender-endpoint* and set its value to *http://localhost:14268/api/traces*.
+
+[source,ruby]
+--
+[standalone@localhost:9990 /] /subsystem=microprofile-opentracing-smallrye/jaeger-tracer=jaeger:write-attribute(name="sender-endpoint", value="http://localhost:14268/api/traces")
+{
+    "outcome" => "success",
+    "response-headers" => {
+        "operation-requires-reload" => true,
+        "process-state" => "reload-required"
+    }
+}
+[standalone@localhost:9990 /] :reload
```

Comment:
s/:reload/reload

Reply:
And I believe the CLI response would be different once you use the command and not the low level op.